spyder/plugins/editor/widgets/tests/test_introspection.py

```diff
@@ -85,6 +109,112 @@ def test_hide_widget_completion(lsp_codeeditor, qtbot):
+    stop = False
+    while not stop:
+        try:
+            with qtbot.waitSignal(completion.sig_show_completions,
+                                  timeout=5000) as sig:
+                pass
+            code_editor.completion_widget.hide()
+        except Exception:
+            stop = True
+
+    with qtbot.waitSignal(completion.sig_show_completions,
+                          timeout=10000) as sig:
+        qtbot.keyClicks(code_editor, ' ')
+
+    assert "os" in [x['label'] for x in sig.args[0]]
+
+    with qtbot.waitSignal(completion.sig_show_completions,
+                          timeout=10000) as sig:
+        qtbot.keyClicks(code_editor, 'n')
+
+    assert "ntpath" in [x['label'] for x in sig.args[0]]
+
+    with qtbot.waitSignal(completion.sig_show_completions,
+                          timeout=10000) as sig:
+        qtbot.keyClicks(code_editor, 'ump')
+
+    assert "numpy" in [x['label'] for x in sig.args[0]]
+
+    with qtbot.waitSignal(completion.sig_show_completions,
+                          timeout=10000) as sig:
+        qtbot.keyPress(code_editor, Qt.Key_Backspace)
+
+    assert "numbers" in [x['label'] for x in sig.args[0]]
+
+    with qtbot.waitSignal(completion.sig_show_completions,
+                          timeout=10000) as sig:
+        qtbot.keyClicks(code_editor, 'py')
+
+    # Due to automatic completion, the completion widget may appear before
+    stop = False
+    while not stop:
+        try:
+            with qtbot.waitSignal(completion.sig_show_completions,
+                                  timeout=5000) as sig:
+                pass
+            code_editor.completion_widget.hide()
+        except Exception:
+            stop = True
+
+    with qtbot.waitSignal(completion.sig_show_completions,
+                          timeout=10000) as sig:
+        qtbot.keyClicks(code_editor, ' ')
+
+    assert "import" in [x['label'] for x in sig.args[0]]
+
+    with qtbot.waitSignal(completion.sig_show_completions,
+                          timeout=10000) as sig:
+        qtbot.keyClicks(code_editor, 'imp')
+
+    with qtbot.waitSignal(completion.sig_show_completions,
+                          timeout=10000) as sig:
+        qtbot.keyPress(code_editor, Qt.Key_Tab)
+
+    assert code_editor.toPlainText() == 'from numpy import'
+
+    # Due to automatic completion, the completion widget may appear before
+    stop = False
+    while not stop:
+        try:
+            with qtbot.waitSignal(completion.sig_show_completions,
+                                  timeout=5000) as sig:
+                pass
+            code_editor.completion_widget.hide()
+        except Exception:
+            stop = True
+
+    with qtbot.waitSignal(completion.sig_show_completions,
+                          timeout=10000) as sig:
+        qtbot.keyClicks(code_editor, ' ')
+
+    assert "random" in [x['label'] for x in sig.args[0]]
+
+
 @pytest.mark.slow
 @pytest.mark.first
 def test_completions(lsp_codeeditor, qtbot):
```

Comment:
Please turn off on-the-fly completions for this test because it makes it way too complex. That would avoid the lot of `while` loops you added to make it pass.